### PR TITLE
add terraform config for an OS instance where irobot will be deployed

### PR DESCRIPTION
**Pull request by sm0179**

### CODE REVIEW

File: terraform/modules/openstack/main.tf [hunk 1]
```diff
@@ -313,6 +313,19 @@ resource "openstack_compute_secgroup_v2" "nfs-server" {
   }
 }
 
+resource "openstack_compute_secgroup_v2" "iRods" {
```
Comment (colin-nolan) [inline terraform/modules/openstack/main.tf:316]: Non-standard name (lowercase only).

File: terraform/modules/openstack/main.tf [hunk 2]
```diff
@@ -313,6 +313,19 @@ resource "openstack_compute_secgroup_v2" "nfs-server" {
...
+  provider    = "openstack"
+  name        = "iRods_${var.region}_${var.env}"
```
Comment (colin-nolan) [inline terraform/modules/openstack/main.tf:318]: Same comment as above on name case.

File: terraform/modules/irobot/main.tf [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+  network {
+    uuid           = "${var.network_id}"
+    access_network = true
+  }
+
+  user_data = "#cloud-config\nhostname: irobot-${count.index + 1}\nfqdn: irobot-${count.index + 1}.${var.domain}"
+
+  metadata = {
+    ansible_groups = "dockerers"
```
Comment (colin-nolan) [inline terraform/modules/irobot/main.tf:55]: You'll need to add the Ansible group for iRobot.

File: terraform/modules/irobot/main.tf [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+  image_name  = "${var.image["name"]}"
+  flavor_name = "${var.flavour}"
+  key_pair    = "${var.key_pair_ids["mercury"]}"
+
+  security_groups = [
+    "${var.security_group_ids["ping"]}",
+    "${var.security_group_ids["ssh"]}",
+    "${var.security_group_ids["http"]}",
+    "${var.security_group_ids["iRods"]}",
+  ]
```
Comment (colin-nolan) [inline terraform/modules/irobot/main.tf:45]: The setup will require access for Kerberos but that can be added when you get there.

**Review by colin-nolan — CHANGES_REQUESTED** [review]
Looks good; just some minor comments.

**Comment by colin-nolan** [comment]
Note: an accepted PR needs to get to GitLab.

File: terraform/modules/openstack/main.tf (hunk 1 above)
Comment (sm0179) [reply terraform/modules/openstack/main.tf:316]: thanks, will change to lowercase everywhere

File: terraform/modules/irobot/main.tf (hunk 4 above)
Comment (sm0179) [reply terraform/modules/irobot/main.tf:45]: yes, I thought about that. setting up Kerberos auth is a different task, not sure if I'm meant to do it

**Pull request closed without merging**